drake/solvers/NonlinearProgramSnoptmex.cpp

```diff
@@ -60,6 +69,16 @@ void mexFunction(int nlhs, mxArray* plhs[], int nrhs, const mxArray* prhs[])
   {
     mexErrMsgIdAndTxt("Drake:NonlinearProgramSnoptmex:InvalidInputs","The usage is [x,F,info,xmul,Fmul] = NonlinearProgramSnoptmex(x,xlow,xupp,Flow,Fupp,userfun,ObjAdd,ObjRow,A,iAfun,jAvar,iGfun,jGvar,options)");
   }
+
+  if (lenrw==0) { // then initialize (sninit needs some default allocation)
+    lenrw = 500000;
+    rw.reset(new snopt::doublereal[lenrw]);
+    leniw = 500000;
+    iw.reset(new snopt::integer[leniw]);
+    lencw = 500;
```

Comment:
we need either the 8* or a /8 in places, because snopt takes it in different formats in different places.  i've taken a pass at cleaning it up before and gave up.  @hongkai-dai has too, i believe.  i agree it's gross, but will leave it as is for now.